File: 1-two-sum/two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        dict = {}
        for i in range(len(nums)):
            dif = target-nums[i]
            if dif in dict:
                return [dict[dif],i]
            dict[nums[i]] = i

File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> unix = new HashMap<>();

        for(int i=0; i<nums.length; i++){
            if (unix.containsKey(target-nums[i])){

                return new int[]{unix.get(target-nums[i]), i};
            }
            unix.put(nums[i],i);

        }
        throw new IllegalArgumentException("No two sum solution");
    }
}